src/test/java/hudson/plugins/git/util/DefaultBuildChooserTest.java

```diff
@@ -30,6 +44,61 @@ public void testChooseGitRevisionToBuildByShaHash() throws Exception {
         candidateRevisions = buildChooser.getCandidateRevisions(false, "aaa" + shaHashCommit1.substring(3), testGitClient, null, null, null);
         assertTrue(candidateRevisions.isEmpty());
     }
+
+    /**
+     * Regression test for a bug that accidentally resulted in empty build candidates.
+     *
+     * This test creates two repositories, one fake-remote repository in temporary directory "dir".
+     * A second repository in temporary directory "dir2".
+     * We create the following commit graph with 2 commits and 3 tags in "dir":
+     * commit1(tag/a)---commit2(tag/b, tag/c)
+     * We then clone this repository into "dir2" and fetch the remote tags as refs with the given refspec.
+     * This is necessary to make the GitClient recognize the tags via getRemoteBranches.
+     *
+     * Expected output if only tag/a has previously been build:
+     * The candidates should only include the commit pointed to by tag/b or tag/c.
+     *
+     * @throws Exception
```

Comment:
if there is no description, then this line is redundant. Just keep only description